test/system/src/test/java/io/pravega/test/system/ControllerRestApiTest.java

```diff
@@ -98,7 +103,9 @@ public ControllerRestApiTest() {
     public static void setup() throws InterruptedException, MarathonException, URISyntaxException {
 
         //1. check if zk is running, if not start it
-        Service zkService = new ZookeeperService("zookeeper");
+        Service zkService = Utils.isDockerLocalExecEnabled()
```

Comment:
This and the next few classes seem to have the same pattern as the previous one. Perhaps there is a way to share code?

Reply:
done. using factory method. `createServiceInstance` for the same.